File: backend/FluffitMember/src/main/java/com/ssafy/fluffitmember/_common/error/ErrorResponse.java
package com.ssafy.fluffitmember._common.error;

import lombok.Builder;
import lombok.Getter;

@Getter
public class ErrorResponse {
    private final int status;
    private final String msg;

    @Builder
    private ErrorResponse(int status, String msg) {
        this.status = status;
        this.msg = msg;
    }
    public static ErrorResponse from(ErrorType errorType) {
        return ErrorResponse.builder()
                .status(errorType.getStatus())
                .msg(errorType.getMsg())
                .build();
    }
}

File: backend/FluffitFlupet/src/main/kotlin/com/ssafy/fluffitflupet/error/ErrorResponse.kt
package com.ssafy.fluffitflupet.error

import org.springframework.http.HttpStatus

class ErrorResponse (
    val statusCode: Int, //http status code
    val msg: String //error message
) {
    companion object {
        fun from(errorType: ErrorType): ErrorResponse {
            return ErrorResponse(
                statusCode = errorType.status.value(),
                msg = errorType.msg
            )
        }

        fun of(status: HttpStatus, msg: String?): ErrorResponse {
            return ErrorResponse(
                statusCode = status.value(),
                msg = msg ?: status.reasonPhrase
            )
        }
    }
}
